lib/active_remote/query_attributes.rb

```diff
@@ -21,25 +21,25 @@ module QueryAttributes
       attribute_method_suffix "?"
     end
 
-    # Test the presence of an attribute
-    #
-    # See {Typecasting::BooleanTypecaster.call} for more details.
-    #
-    # @example Query an attribute
-    #   person.query_attribute(:name)
-    #
-    def query_attribute(name)
-      if respond_to?("#{name}?")
-        send("#{name}?")
+    def query_attribute(attr_name)
+      value = self[attr_name]
+
+      case value
+      when true        then true
+      when false, nil  then false
       else
-        raise ::ActiveRemote::UnknownAttributeError, "unknown attribute: #{name}"
+        if value.respond_to?(:zero?)
+          !value.zero?
+        else
+          !value.blank?
+        end
       end
     end
 
   private
 
-    def attribute?(name)
-      ::ActiveRemote::Typecasting::BooleanTypecaster.call(read_attribute(name))
```

Comment:
Removing this typecaster reference is the primary driver behind this change. Since the behavior differed from that of ActiveRecord, I decided to bring it inline. The only difference is that strings such as `'FALSE'` and `'0.0'` will now be considered present, rather than absent, but that seemed it should have always been.